sdm/src/main/java/com/sap/cds/sdm/configuration/Registration.java

```diff
@@ -0,0 +1,127 @@
+import com.sap.cds.feature.attachments.handler.common.AttachmentsReader;
+import com.sap.cds.feature.attachments.handler.common.DefaultAssociationCascader;
+import com.sap.cds.feature.attachments.handler.common.DefaultAttachmentsReader;
+import com.sap.cds.feature.attachments.handler.draftservice.DraftCancelAttachmentsHandler;
+import com.sap.cds.feature.attachments.handler.draftservice.DraftPatchAttachmentsHandler;
+import com.sap.cds.feature.attachments.handler.draftservice.modifier.ActiveEntityModifier;
+import com.sap.cds.feature.attachments.service.AttachmentService;
+import com.sap.cds.feature.attachments.service.handler.transaction.EndTransactionMalwareScanProvider;
+import com.sap.cds.feature.attachments.service.handler.transaction.EndTransactionMalwareScanRunner;
+import com.sap.cds.feature.attachments.service.malware.AsyncMalwareScanExecutor;
+import com.sap.cds.feature.attachments.service.malware.DefaultAttachmentMalwareScanner;
+import com.sap.cds.feature.attachments.service.malware.client.DefaultMalwareScanClient;
+import com.sap.cds.feature.attachments.service.malware.client.httpclient.MalwareScanClientProviderFactory;
+import com.sap.cds.feature.attachments.service.malware.client.mapper.DefaultMalwareClientStatusMapper;
+import com.sap.cds.feature.attachments.service.malware.constants.MalwareScanConstants;
+import com.sap.cds.feature.attachments.utilities.LoggingMarker;
+import com.sap.cds.services.environment.CdsProperties.ConnectionPool;
+import com.sap.cds.services.handler.EventHandler;
+import com.sap.cds.services.outbox.OutboxService;
+import com.sap.cds.services.persistence.PersistenceService;
+import com.sap.cds.services.runtime.CdsRuntimeConfiguration;
+import com.sap.cds.services.runtime.CdsRuntimeConfigurer;
+import com.sap.cds.services.utils.environment.ServiceBindingUtils;
+import com.sap.cloud.environment.servicebinding.api.ServiceBinding;
+
+/**
+ * The class {@link Registration} is a configuration class that registers the
+ * services and event handlers for the attachments feature.
+ */
+public class Registration implements CdsRuntimeConfiguration {
+
+	private static final Logger logger = LoggerFactory.getLogger(Registration.class);
+	private static final Marker marker = LoggingMarker.ATTACHMENT_SERVICE_REGISTRATION.getMarker();
+
+	@Override
+	public void services(CdsRuntimeConfigurer configurer) {
+		configurer.service(buildAttachmentService());
+	}
+
+	@Override
+	public void eventHandlers(CdsRuntimeConfigurer configurer) {
+		logger.info(marker, "Registering event handler for attachment service");
+		CacheConfig.initializeCache();
+		var attachmentService = configurer.getCdsRuntime().getServiceCatalog().getService(AttachmentService.class,
+				AttachmentService.DEFAULT_NAME);
+		var outbox = configurer.getCdsRuntime().getServiceCatalog().getService(OutboxService.class,
+				OutboxService.PERSISTENT_UNORDERED_NAME);//need to check if required
```

Comment:
IS this needed?

Reply:
this is required now as for a method its mandatory can be removed later